<?xml version="1.0"?>
<launch>
  <arg name="output" default="screen"/>
  <arg name="goal_topic" default="/move_base_simple/goal"/>
  <arg name="regist_goal_topic" default="/clicked_point"/>
  <arg name="config_file" default="$(find waypoint_server)/config/waypoint_server.yaml"/>
  <group ns="waypoint_manager">
    <node pkg="waypoint_server" type="waypoint_server_node" name="waypoint_server" output="$(arg output)">
      <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
      <remap from="waypoint/regist_point" to="$(arg regist_goal_topic)"/>
      <param name="clear_costmap_srv" value="/move_base/clear_costmaps" />
    </node>
    <node pkg="waypoint_server" type="waypoint_to_posestamped_node" name="waypoint_to_posestamped" output="$(arg output)">
      <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
      <remap from="move_base_simple/goal" to="$(arg goal_topic)"/>
    </node>
    <node pkg="waypoint_visualization" type="waypoint_visualization_node" name="waypoint_visualization" output="$(arg output)">
      <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
      <remap from="waypoint/regist_point" to="$(arg regist_goal_topic)"/>
    </node>
    <node pkg="goal_event_handler" type="radius_node" name="radius_node" output="$(arg output)">
      <rosparam command="load" file="$(arg config_file)" subst_value="true"/>
    </node>
  </group>
</launch>
